rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -889,25 +904,18 @@ def _maybe_decode_erc20_approve(
             action_items: list[ActionItem],  # pylint: disable=unused-argument
             all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
     ) -> DecodingOutput:
-        if tx_log.topics[0] != ERC20_APPROVE:
+        if (
+            tx_log.topics[0] != ERC20_OR_ERC721_APPROVE or
+            (token_kind := self._get_transfer_or_approval_token_kind(tx_log=tx_log)) is None
+        ):
             return DEFAULT_DECODING_OUTPUT
 
-        if token is None:
-            try:
-                token = get_or_create_evm_token(
-                    userdb=self.database,
-                    evm_address=tx_log.address,
-                    chain_id=self.evm_inquirer.chain_id,
-                    token_kind=EvmTokenKind.ERC20,
-                    evm_inquirer=self.evm_inquirer,
-                    encounter=TokenEncounterInfo(tx_hash=transaction.tx_hash),
-                )
-            except NotERC20Conformant:
-                return DEFAULT_DECODING_OUTPUT  # ignore non-ERC20 approval for now
-
-        if len(tx_log.topics) == 3:
+        collectible_id = None
+        if len(tx_log.topics) in (3, 4):
             owner_address = bytes_to_address(tx_log.topics[1])
             spender_address = bytes_to_address(tx_log.topics[2])
+            if len(tx_log.topics) == 4:
+                collectible_id = str(int.from_bytes(tx_log.topics[3]))
```

Comment:
```suggestion
            collectible_id = str(int.from_bytes(tx_log.topics[3])) if len(tx_log.topics) == 4 else None
```
I believe is not possible to avoid checking twice unless you unroll and duplicate... what doesn't sound bad to me tbh but and if check is not too much